### ISSUE TITLE
Failed/aborted - netdata-installer.sh exited with error 

### ISSUE THREAD

**Issue by verfai** [netdata/netdata#9579]
##### Bug report summary
New to Netdata, great project. Just tried installing on 3 similar CentOS 7 servers, 1 installed successfully, the other 2 fail.
```
<...>
  CC       cli/cli.o
  CCLD     apps.plugin
  CCLD     cgroup-network
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_object__add_program':
libbpf.c:(.text+0xd65): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_object__elf_collect':
libbpf.c:(.text+0x68ce): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_object__collect_externs':
libbpf.c:(.text+0x6fd1): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_core_find_cands':
libbpf.c:(.text+0xa1da): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_program__reloc_text':
libbpf.c:(.text+0xbd3d): undefined reference to `reallocarray'
collect2: error: ld returned 1 exit status
make[2]: *** [cgroup-network] Error 1
make[2]: *** Waiting for unfinished jobs....
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_object__add_program':
libbpf.c:(.text+0xd65): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_object__elf_collect':
libbpf.c:(.text+0x68ce): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_object__collect_externs':
libbpf.c:(.text+0x6fd1): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_core_find_cands':
libbpf.c:(.text+0xa1da): undefined reference to `reallocarray'
externaldeps/libbpf/libbpf.a(libbpf.o): In function `bpf_program__reloc_text':
libbpf.c:(.text+0xbd3d): undefined reference to `reallocarray'
collect2: error: ld returned 1 exit status
make[2]: *** [apps.plugin] Error 1
make[2]: Leaving directory `/home/netdata-kickstart-lTUBXF/netdata-v1.23.2-32-g1602d66a'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/home/netdata-kickstart-lTUBXF/netdata-v1.23.2-32-g1602d66a'
make: *** [all] Error 2
 FAILED   

 FAILED  

 ABORTED  netdata-installer.sh exited with error 
```

##### OS / Environment

```
/etc/centos-release:CentOS Linux release 7.8.2003 (Core)
/etc/os-release:NAME="CentOS Linux"
/etc/os-release:VERSION="7 (Core)"
/etc/os-release:ID="centos"
/etc/os-release:ID_LIKE="rhel fedora"
/etc/os-release:VERSION_ID="7"
/etc/os-release:PRETTY_NAME="CentOS Linux 7 (Core)"
/etc/os-release:ANSI_COLOR="0;31"
/etc/os-release:CPE_NAME="cpe:/o:centos:centos:7"
/etc/os-release:HOME_URL="https://www.centos.org/"
/etc/os-release:BUG_REPORT_URL="https://bugs.centos.org/"
/etc/os-release:
/etc/os-release:CENTOS_MANTISBT_PROJECT="CentOS-7"
/etc/os-release:CENTOS_MANTISBT_PROJECT_VERSION="7"
/etc/os-release:REDHAT_SUPPORT_PRODUCT="centos"
/etc/os-release:REDHAT_SUPPORT_PRODUCT_VERSION="7"
/etc/os-release:
/etc/redhat-release:CentOS Linux release 7.8.2003 (Core)
/etc/system-release:CentOS Linux release 7.8.2003 (Core)
```

Thanks for your help!

**Comment by vlvkobal** [comment]
Could you run the following script and check its return code (`./script.sh | echo $?`)
```
#!/bin/sh

tfile=$(mktemp /tmp/test_reallocarray_XXXXXXXX.c)
ofile=${tfile%.c}.o

cat > $tfile <<EOL
#define _GNU_SOURCE
#include <stdlib.h>
int main(void)
{
	return !!reallocarray(NULL, 1, 1);
}
EOL

gcc $tfile -o $ofile >/dev/null 2>&1
if [ $? -ne 0 ]; then echo "FAIL"; fi
/bin/rm -f $tfile $ofile
```

**Comment by verfai** [comment]
Thanks for your reply;

`[root@b1 test]# ./script.sh | echo $?
0
-bash: ./script.sh: Permission denied`

**Comment by mfundul** [comment]
You can try:
```
chmod +x script.sh
./script.sh
echo $?
```

**Comment by verfai** [comment]
```
[root@backup1 test]# chmod +x script.sh
[root@backup1 test]# ./script.sh
FAIL
[root@backup1 test]# echo $?
0
[root@backup1 test]# 
```

**Comment by vlvkobal** [comment]
@thiagoftsm I remember you can reproduce something like this on your VM. Could you check if libbpf is built with `-DCOMPAT_NEED_REALLOCARRAY`?

**Comment by thiagoftsm** [comment]
> @thiagoftsm I remember you can reproduce something like this on your VM. Could you check if libbpf is built with `-DCOMPAT_NEED_REALLOCARRAY`?

I tested now `-DCOMPAT_NEED_REALLOCARRAY` and `-UCOMPAT_NEED_REALLOCARRAY` and I could not compile in any situation on CentOS 7.8.2003

**Comment by vlvkobal** [comment]
It looks like tar doesn't extract file permissions in the present case (`check-reallocarray.sh: Permission denied`)
```
[/tmp/netdata-libbpf-RtER8C/libbpf-0.0.9/src]# env CFLAGS= CXXFLAGS= LDFLAGS= BUILD_STATIC_ONLY=y OBJDIR=build DESTDIR=.. make install 
make: execvp: ../scripts/check-reallocarray.sh: Permission denied
mkdir -p build/staticobjs
cc -I. -I../include -I../include/uapi  -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64    -c bpf.c -o build/staticobjs/bpf.o
```
We need to check it though.

**Comment by bbbenjie** [comment]
Looks like /tmp is mounted with noexec option? This would explain starting ```check-reallocarray.sh``` within /tmp/netdata-whatever fails with ```Permission denied```.
Same situation and problem on one of my servers starting with the nightly update on 23/07/2020.

```
Do 23. Jul 04:34:27 CEST 2020 : INFO:  Current Version: 001023002024
Do 23. Jul 04:34:27 CEST 2020 : INFO:  Latest Version: 001023002044
```

**Comment by jrod9k1** [comment]
@bbbenjie spot on, thank you for this. I ran into this issue on a CentOS 7 cPanel server that mounts /tmp with noexec. Temporarily remounting without noexec did the trick, kickstart script ran without a hitch
`mount /tmp -o remount,rw`

**Comment by Ferroin** [comment]
I've opened another issue for the `/tmp` `noexec` thing: #9626

**Comment by prologic** [comment]
I'm going to close this issue as this can be fixed with a PR for #9626 Thanks @Ferroin

**Comment by UltimateByte** [comment]
Hey,

I additionally had to do rm -rf /tmp/netdata* to be able to install, after remounting /tmp with write permissions on my CentOS 7.

Hope it helps somebody :)
